nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -58,3 +71,389 @@ def run(self, published_after, debug=False):  # pylint: disable=too-many-locals,
 
         diff = VulnerabilityLCM.objects.count() - count_before
         self.logger.info("Processed %d CVEs and generated %d Vulnerabilities." % (cves.count(), diff))
+
+
+class NistCveSyncSoftware(Job):
+    """Checks all device SoftwareVersion for NIST recorded vulnerabilities."""
+
+    name = "NIST - Software CVE Search"
+    description = "Searches the NIST DBs for CVEs related to SoftwareVersion"
+    read_only = False
+
+    nist_integration = ObjectVar(
+        model=ExternalIntegration,
+        query_params={
+            "name__contains": "NIST",
+        },
+        description="Select the NIST integration to use",
+        required=True,
+        default=lambda: ExternalIntegration.objects.get(name="NAUTOBOT DLM NIST EXTERNAL INTEGRATION"),
+    )
+
+    class Meta:  # pylint: disable=too-few-public-methods
+        """Meta class for the job."""
+
+        commit_default = True
+        soft_time_limit = 3600
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the job for use with the NIST API Key."""
+        super().__init__(*args, **kwargs)
+        self.nist_api_key = None
+
+    def run(self, *args, **kwargs: dict):  # pylint: disable=too-many-locals
+        """Check all software in DLC against NIST database and associate registered CVEs.
+
+        Args:
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+        try:
+            self.nist_api_key = (
+                kwargs.get("nist_integration").secrets_group.secrets.get(name="NAUTOBOT DLM NIST API KEY").get_value()
+            )  # pylint: disable=attribute-defined-outside-init
+        except SecretError as err:
+            self.logger.error(
+                "This job REQUIRES a fully configured Secret named 'NAUTOBOT DLM NIST API KEY'.  Please refer to the documentation to obtain an api key and complete configuration. %s",
+                err,
+            )
+            return
+
+        if kwargs.get("nist_integration"):
+            self.integration = kwargs["nist_integration"]  # pylint: disable=attribute-defined-outside-init
+        else:
+            self.logger.error("NIST ExternalIntegration object is required.")
+            return
+
+        cve_counter = 0
+
+        for software in SoftwareVersion.objects.all():
+            platform = software.platform.network_driver.lower()
+            version = software.version.replace(" ", "")
+            self.logger.info(platform)
```

Comment:
This log message should be more descriptive.

Reply:
I removed it.  This was leftover from troubleshooting and didn't need to be there.